src/bin/miri.rs

```diff
@@ -23,7 +23,7 @@ use syntax::ast;
 use std::path::PathBuf;
 
 struct MiriCompilerCalls {
-    default: RustcDefaultCalls,
+    default: Box<RustcDefaultCalls>,
```

Comment:
Is it not possible to remove the `MiriCompilerCalls` entirely with your new system?

Reply:
No, because of https://github.com/solson/miri/pull/383/files#diff-c3d602c5c8035a16699ce9c015bfeceaR90 which changes the `CompileController` depending on the session, which is not available when `rustc_driver::run_compiler` is called.